homeassistant/components/ambient_network/sensor.py

```diff
@@ -0,0 +1,305 @@
+        device_class=SensorDeviceClass.PRECIPITATION,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=2,
+    ),
+    SensorEntityDescription(
+        key=TYPE_WINDDIR,
+        translation_key="wind_direction",
+        native_unit_of_measurement=DEGREE,
+        suggested_display_precision=0,
+    ),
+    SensorEntityDescription(
+        key=TYPE_WINDGUSTMPH,
+        translation_key="wind_gust",
+        native_unit_of_measurement=UnitOfSpeed.MILES_PER_HOUR,
+        device_class=SensorDeviceClass.WIND_SPEED,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key=TYPE_WINDSPEEDMPH,
+        native_unit_of_measurement=UnitOfSpeed.MILES_PER_HOUR,
+        device_class=SensorDeviceClass.WIND_SPEED,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key=TYPE_YEARLYRAININ,
+        translation_key="yearly_rain",
+        native_unit_of_measurement=UnitOfPrecipitationDepth.INCHES,
+        device_class=SensorDeviceClass.PRECIPITATION,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Ambient Network sensor entities."""
+
+    coordinator: AmbientNetworkDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    await coordinator.async_config_entry_first_refresh()
+    if coordinator.data == {}:
+        raise PlatformNotReady("Error while connecting to Ambient Network")
+
+    if coordinator.config_entry is not None:
+        async_add_entities(
+            AmbientNetworkSensor(
+                coordinator,
+                description,
+                coordinator.config_entry.data[CONF_MAC],
+            )
+            for description in SENSOR_DESCRIPTIONS
+            if coordinator.data.get(description.key) is not None
+        )
+
+
+class AmbientNetworkSensor(AmbientNetworkEntity, SensorEntity):
+    """A sensor implementation for an Ambient Weather Network sensor."""
+
+    def __init__(
+        self,
+        coordinator: AmbientNetworkDataUpdateCoordinator,
+        description: SensorEntityDescription,
+        mac_address: str,
+    ) -> None:
+        """Initialize a sensor object."""
+
+        super().__init__(coordinator, description, mac_address)
+
+    def _update_attrs(self) -> None:
+        """Update sensor attributes."""
+
+        value = self.coordinator.data.get(self.entity_description.key)
+
+        # Treatments for special units.
+        if value is not None and self.device_class == SensorDeviceClass.TIMESTAMP:
+            value = datetime.fromtimestamp(value / 1000, tz=dt_util.DEFAULT_TIME_ZONE)
+
+        self._attr_available = value is not None
```

Comment:
Can single sensors return None and thereby be unavailable?

Reply:
I do not know the answer to this. Since a whether station is made up of a sensor array, it is in theory possible that one of the sensors dies or is unplugged. I do not know how the Ambient firmware would handle that; whether it omits the failing sensor or reports a previous value.